src/ipa_pc/data_structures.rs

```diff
@@ -665,48 +669,34 @@ impl<G: AffineCurve> CanonicalSerialize for BatchProof<G> {
         // Serialize proof
         CanonicalSerialize::serialize(&self.proof, &mut writer)?;
 
-        // Serialize batch_commitment
+        // Serialize h_comm
         // More than enough for practical applications
-        let batch_commitment_len = u8::try_from(self.batch_commitment.len()).map_err(|_| SerializationError::NotEnoughSpace)?;
-        CanonicalSerialize::serialize(&batch_commitment_len, &mut writer)?;
+        let h_comm_len = u8::try_from(self.h_comm.len()).map_err(|_| SerializationError::NotEnoughSpace)?;
+        CanonicalSerialize::serialize(&h_comm_len, &mut writer)?;
 
         // Save only one of the coordinates of the point and one byte of flags in order
         // to be able to reconstruct the other coordinate
-        for comm in self.batch_commitment.iter() {
+        for comm in self.h_comm.iter() {
```

Comment:
Just a side question: Don't we have an implementation of CanonicalSerialize for segmented commitments?

Reply:
Not sure what you mean. Segmented commitment - is just a vector of points, it is not a named structure, so cannot have some specific serialize implementation